File: pkg/codegen/testing/test/testdata/aws-iam-policy-pp/dotnet/aws-iam-policy.cs
using System.Collections.Generic;
using System.Linq;
using System.Text.Json;
using Pulumi;
using Aws = Pulumi.Aws;

return await Deployment.RunAsync(() => 
{
    // Create a policy with multiple Condition keys
    var policy = new Aws.Iam.Policy("policy", new()
    {
        Path = "/",
        Description = "My test policy",
        PolicyDocument = JsonSerializer.Serialize(new Dictionary<string, object?>
        {
            ["Version"] = "2012-10-17",
            ["Statement"] = new[]
            {
                new Dictionary<string, object?>
                {
                    ["Effect"] = "Allow",
                    ["Action"] = "lambda:*",
                    ["Resource"] = "arn:aws:lambda:*:*:function:*",
                    ["Condition"] = new Dictionary<string, object?>
                    {
                        ["StringEquals"] = new Dictionary<string, object?>
                        {
                            ["aws:RequestTag/Team"] = new[]
                            {
                                "iamuser-admin",
                                "iamuser2-admin",
                            },
                        },
                        ["ForAllValues:StringEquals"] = new Dictionary<string, object?>
                        {
                            ["aws:TagKeys"] = new[]
                            {
                                "Team",
                            },
                        },
                    },
                },
            },
        }),
    });

    return new Dictionary<string, object?>
    {
        ["policyName"] = policy.Name,
    };
});

File: pkg/codegen/testing/test/testdata/aws-iam-policy-pp/go/aws-iam-policy.go
package main

import (
	"encoding/json"

	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/iam"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		tmpJSON0, err := json.Marshal(map[string]interface{}{
			"Version": "2012-10-17",
			"Statement": []map[string]interface{}{
				map[string]interface{}{
					"Effect":   "Allow",
					"Action":   "lambda:*",
					"Resource": "arn:aws:lambda:*:*:function:*",
					"Condition": map[string]interface{}{
						"StringEquals": map[string]interface{}{
							"aws:RequestTag/Team": []string{
								"iamuser-admin",
								"iamuser2-admin",
							},
						},
						"ForAllValues:StringEquals": map[string]interface{}{
							"aws:TagKeys": []string{
								"Team",
							},
						},
					},
				},
			},
		})
		if err != nil {
			return err
		}
		json0 := string(tmpJSON0)
		// Create a policy with multiple Condition keys
		policy, err := iam.NewPolicy(ctx, "policy", &iam.PolicyArgs{
			Path:        pulumi.String("/"),
			Description: pulumi.String("My test policy"),
			Policy:      pulumi.String(json0),
		})
		if err != nil {
			return err
		}
		ctx.Export("policyName", policy.Name)
		return nil
	})
}
